src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [fromParts, fromSeparator] = [
+    fromFormat.slice(0, -1),
+    fromFormat.slice(-1)[0],
+  ];
+  const [toParts, toSeparator] = [toFormat.slice(0, -1), toFormat.slice(-1)[0]];
+
+  const dateParts = date.split(fromSeparator);
+  const dateMap = {};
+
+  fromParts.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  const result = toParts.map((part) => {
+    if (part === 'YY' && dateMap['YYYY']) {
+      return dateMap['YYYY'].slice(-2);
+    }
+
+    if (part === 'YYYY' && dateMap['YY']) {
+      const year = parseInt(dateMap['YY'], 10);
+
+      return year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
+    }
```

Comment:
The conversion from 'YY' to 'YYYY' assumes that 'YY' is always present in the `fromFormat`. Consider adding a check to ensure 'YY' exists in `dateMap` before converting.